python/demo/gsoc2022/demo_scattering_boundary_conditions.py

```diff
@@ -0,0 +1,574 @@
+# ---
+# jupyter:
+#   jupytext:
+#     formats: ipynb,py
+#     text_representation:
+#       extension: .py
+#       format_name: light
+#       format_version: '1.5'
+#       jupytext_version: 1.13.8
+#   kernelspec:
+#     display_name: Python 3 (ipykernel)
+#     language: python
+#     name: python3
+# ---
+
+# # Scattering from a wire with scattering boundary conditions
+# This demo is implemented in two files: one for the mesh
+# generation with gmsh, and one for the variational forms
+# and the solver. It illustrates how to:
+#
+# - Use complex quantities in FEniCSx
+# - Setup and solve Maxwell's equations
+# - Implement Scattering Boundary Conditions
+#
+# ## Equations, problem definition and implementation
+#
+# First of all, let's import the modules that will be used:
+
+# +
+import gmsh
+import numpy as np
+import pyvista
+from gmsh_helpers import gmsh_model_to_mesh
+from mesh_wire import generate_mesh_wire
+from utils import calculate_analytical_efficiencies
+
+import ufl
+from dolfinx import fem, plot
+from dolfinx.io import VTXWriter
+from ufl import (FacetNormal, as_vector, conj, cross, curl, inner, lhs, rhs,
+                 sqrt)
+
+from mpi4py import MPI
+from petsc4py import PETSc
+
+# -
+
+# Since we want to solve Maxwell's equation, we need to specify that
```

Comment:
I'd suggest using `time-harmonic Maxwell equations` or `frequency domain` 

We can solve Maxwell equations with real mode for time depedent problems.